src/lsm/level_data_iterator.zig

```diff
@@ -10,37 +10,43 @@ const ManifestType = @import("manifest.zig").ManifestType;
 const alloc_block = @import("grid.zig").alloc_block;
 const GridType = @import("grid.zig").GridType;
 const TableDataIteratorType = @import("table_data_iterator.zig").TableDataIteratorType;
-const LevelIndexIteratorType = @import("level_index_iterator.zig").LevelIndexIteratorType;
 
-/// A LevelDataIterator iterates the data blocks of every table in a key range in ascending key order.
-pub fn LevelDataIteratorType(comptime Table: type, comptime Storage: type) type {
+// Iterates over the data blocks in a level B table. References to the level B
+// tables to be iterated over are passed via the `tables` field in the context,
+// in the start() function.
+// * Assumption: This iterator works under the assumption that the references to
+// the tables are stable. For references fetched from the ManifestLevel, this
+// assumption only holds true till move_table & insert_table are called as a part
+// of apply_to_manifest. This is because move_table & insert_table are both operations
+// that insert/remove tables from the ManifestLevel segmented arrays, which could cause
+// rearrangements in the memory layout of the arrays. These rearrangements could
+// cause our table references to get invalidated.
+pub fn LevelIteratorType(comptime Table: type, comptime Storage: type) type {
```

Comment:
Rename the file to `level_iterator.zig`

Reply:
Hey, Rafael! As part of https://github.com/tigerbeetledb/tigerbeetle/pull/875#discussion_r1230204218, we decided to defer file rename to a separate PR, because GitHub shows the rename as a file deletion + file addition, which makes reviewing cumbersome...